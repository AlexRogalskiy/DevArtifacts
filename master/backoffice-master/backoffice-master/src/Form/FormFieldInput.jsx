import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import {
  TextField,
  InputAdornment,
  MenuItem,
} from '@material-ui/core'

const InputEnd = ({ icon }) => (
  <Fragment>
    {icon ? (
      <InputAdornment position="end">
        {icon}
      </InputAdornment>
    ) : null}
  </Fragment>
)

InputEnd.propTypes = {
  icon: PropTypes.element,
}

InputEnd.defaultProps = {
  icon: null,
}

const FormFieldInput = ({
  id,
  type,
  title,
  value,
  isMultiline,
  handleChange,
  helperText,
  defaultValue,
  select,
  options,
  rows,
  isRequired,
  error,
  iconEnd,
  isDisabled,
  classNames,
  onFocus,
  onBlur,
  onKeyPress,
}) => (
  <TextField
    id={id}
    type={type}
    label={title}
    value={value}
    multiline={isMultiline}
    onChange={handleChange(id)}
    className={classNames.join(' ')}
    helperText={helperText}
    defaultValue={defaultValue}
    select={select}
    required={isRequired}
    error={error}
    margin="normal"
    fullWidth
    rows={rows}
    disabled={isDisabled}
    onFocus={onFocus}
    onBlur={onBlur}
    onKeyPress={onKeyPress}
    InputProps={{
      endAdornment: (<InputEnd icon={iconEnd} />),
    }}
  >
    {options ?
        options.map(option => (
          <MenuItem value={option} key={option.toLowerCase().replace(/ /, '')}>
            {option}
          </MenuItem>
        ))
      : null
    }
  </TextField>
)

FormFieldInput.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.number,
  ]),
  defaultValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.number,
  ]),
  rows: PropTypes.number,
  isMultiline: PropTypes.bool,
  helperText: PropTypes.string,
  isRequired: PropTypes.bool,
  error: PropTypes.bool,
  classNames: PropTypes.arrayOf(PropTypes.string),
  handleChange: PropTypes.func,
  select: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.string),
  isDisabled: PropTypes.bool,
  iconEnd: PropTypes.element,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onKeyPress: PropTypes.func,
}

FormFieldInput.defaultProps = {
  type: 'text',
  value: '',
  defaultValue: undefined,
  isMultiline: false,
  helperText: '',
  classNames: [],
  isRequired: false,
  error: false,
  rows: 1,
  select: false,
  options: [],
  iconEnd: null,
  isDisabled: false,
  handleChange: () => { },
  onFocus: () => { },
  onBlur: () => { },
  onKeyPress: () => { },
}

export default FormFieldInput
