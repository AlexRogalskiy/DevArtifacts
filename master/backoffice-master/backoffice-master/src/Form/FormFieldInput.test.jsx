import React from 'react'
import { mount } from 'enzyme'

import FormFieldInput from './FormFieldInput'

it('renders correctly', () => {
  const tree = mount((
    <FormFieldInput />
  ))

  expect(tree).toMatchSnapshot()
})


it('renders disabled', () => {
  const tree = mount((
    <FormFieldInput
      isDisabled
    />
  ))

  expect(tree).toMatchSnapshot()
})
