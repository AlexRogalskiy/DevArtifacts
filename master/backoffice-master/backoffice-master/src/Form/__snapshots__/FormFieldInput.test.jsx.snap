// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  onKeyPress={[Function]}
>
  <div
    className="MuiInput-root-5 MuiInput-fullWidth-12 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      rows={1}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`renders disabled 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  onKeyPress={[Function]}
>
  <div
    className="MuiInput-root-5 MuiInput-disabled-8 MuiInput-fullWidth-12 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13 MuiInput-disabled-8"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      rows={1}
      type="text"
      value=""
    />
  </div>
</div>
`;
